mhlib/app.go

```diff
@@ -294,64 +294,70 @@ func (a *App) render(configFile string) (*string, *string, *[]byte, error) {
 	return &chart, &chartVersion, &overrides, nil
 }
 
-
-func selfRender(templateValuesStr string, enableGomplate bool) (string, error) {
+func selfRender(templateValuesStr string) (string, error) {
 	/*
-	This function will accept an input string and run it through the
-	  templating engine as both the values dictionary as well as the
-	  template string.  It will repeat the process until the templating
-	  result stops changing.
+		This function will accept an input string and run it through the
+		  templating engine as both the values dictionary as well as the
+		  template string.  It will repeat the process until the templating
+		  result stops changing.
 	*/
 
-	type Values map[string]interface{}
+	type gomplateConfig struct {
+		Gomplate gomplate.Config `yaml:"gomplate,omitempty"`
+	}
+	var gomp gomplateConfig
+	err := yaml.Unmarshal([]byte(templateValuesStr), &gomp)
+	if err != nil {
+		return "", err
+	}
+
+	funcs := template.FuncMap{
```

Comment:
Are you adding a function?  Perhaps we shuld upstream this function into gomplate instead?  The guy is really responsive.

Reply:
I am not sure I am using this function anymore.  I could remove.  This should make struct elements intensive however this is no longer a struct,  I think it probably should be, however the structure of our yaml does not support this.